### PR TITLE
[bug fix][multi pdn]Fix for multi pdn tunnel creation

### PR DESCRIPTION
**Pull request by pruthvihebbani**
Summary:
Modify bearer request was not sent from MME to SGW for secondary PDN connection due to which gtp tunnel was not created.This PR fixes the issue

Test Plan:
1.	Executed S1 SIM sanity
2.	Executed multi PDN TCs

### CODE REVIEW

File: lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c [hunk 1]
```diff
@@ -753,7 +754,13 @@ esm_cause_t esm_recv_activate_default_eps_bearer_context_accept(
   if (rc != RETURNerror) {
     esm_cause = ESM_CAUSE_SUCCESS;
   }
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c]: This is not part of your diff, but is this section needed? On line 703, esm_cause is set to ESM_CAUSE_SUCCESS by default. The function call esm_proc_default_eps_bearer_context_accept(emm_context, ebi, &esm_cause) modifies it conditionally. I think if rc != RETURNok condition should be checked and if true, an error cause should be returned.

File: lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c [hunk 2]
```diff
@@ -753,7 +754,13 @@ esm_cause_t esm_recv_activate_default_eps_bearer_context_accept(
...
-
+  /* If activate default EPS bearer context accept message is received for a
+   * new standalone PDN connection, send modify bearer request to sgw
+   */
+  if (emm_context->esm_ctx.is_standalone == true) {
+    emm_context->esm_ctx.is_standalone = false;
+    send_modify_bearer_req(ue_id, ebi);
+  }
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c]: should not we check if rc == RETURNok ?

**Review by ulaskozat — CHANGES_REQUESTED** [review]

File: lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c (hunk 1 above)
Comment (pruthvihebbani) [reply lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c]: fixed

File: lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c (hunk 2 above)
Comment (pruthvihebbani) [reply lte/gateway/c/oai/tasks/nas/esm/sap/esm_recv.c]: fixed

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 3]
```diff
@@ -84,6 +84,119 @@
...
+    &mme_app_desc_p->mme_ue_contexts, ue_id);
+
+  if (ue_context_p == NULL) {
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "ue_context_p is NULL, did not send S11_MODIFY_BEARER_REQUEST\n");
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
+  }
+  MessageDef* message_p =
+    itti_alloc_new_message(TASK_MME_APP, S11_MODIFY_BEARER_REQUEST);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:104]: When we return due to errors before actually sending the itti message, we should free the allocated memory for the itti message (please see comments below)

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 4]
```diff
@@ -84,6 +84,119 @@
...
+  if (message_p == NULL) {
+    OAILOG_ERROR(
+      LOG_MME_APP, "Cannot allocate memory to S11_MODIFY_BEARER_REQUEST\n");
+    unlock_ue_contexts(ue_context_p);
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:109]: no need to free as memory allocation itself has failed.

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 5]
```diff
@@ -84,6 +84,119 @@
...
+    mme_app_get_bearer_context(ue_context_p, ebi);
+
+  if (bearer_cntxt == NULL) {
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "Bearer context is null, did not send S11_MODIFY_BEARER_REQUEST for ebi"
+      "%u\n",
+      ebi);
+    unlock_ue_contexts(ue_context_p);
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:129]: we need to free the itti message as none is being sent.

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 6]
```diff
@@ -84,6 +84,119 @@
...
+
+  s11_modify_bearer_request->bearer_contexts_to_be_modified
+    .bearer_contexts[item]
+    .s1_eNB_fteid.interface_type = S1_U_ENODEB_GTP_U;
+  pdn_cid_t cid = ue_context_p->bearer_contexts[EBI_TO_INDEX(ebi)]->pdn_cx_id;
+  pdn_context_t* pdn_context_p = ue_context_p->pdn_contexts[cid];
+  if (pdn_context_p == NULL) {
+    OAILOG_ERROR(LOG_MME_APP, "Did not find PDN context for ebi %u\n", ebi);
+    unlock_ue_contexts(ue_context_p);
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:147]: need to free the itti message.

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 7]
```diff
@@ -84,6 +84,119 @@
...
+      .bearer_contexts[item]
+      .s1_eNB_fteid.ipv6 = 1;
+    memcpy(
+      &s11_modify_bearer_request->bearer_contexts_to_be_modified
+         .bearer_contexts[item]
+         .s1_eNB_fteid.ipv6_address,
+      &bearer_cntxt->enb_fteid_s1u.ipv6_address,
+      sizeof(bearer_cntxt->enb_fteid_s1u.ipv6_address));
+  } else {
+    OAILOG_ERROR(LOG_MME_APP, "Unknown IP address\n");
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:173]: do we still send modify bearer if IP address is unknown?

**Pull request closed without merging**